src/httpfetch.cpp

```diff
@@ -678,7 +679,7 @@ class CurlFetchThread : public Thread
 				that the thread should be stopped.)
 			*/
 			if (m_all_ongoing.empty())
-				waitForRequest(100000000);
+				waitForRequest(5000);
```

Comment:
why do we need this?

Reply:
I unfortunately don't know, undoing the change doesn't seem to make any difference.

Reply:
This might have interacted with the now removed `FreeThread` but I don't see a reason for it now.
Please change it back.